Parameterization: cleanup

afabri opened the pull request:
## Summary of Changes

LSCM parameterization was prepared to fall back to code forked from OpenNL some years back.  As it is 10 times slower, we better depend on Eigen just as the other parameterization methods do.   

### Todo

- [x] Currently one file remains, but that fills Eigen matrices and calls an Eigen solver.   No need to be in a namespace OpenNL. 
- [x] cleanup usage.hhml
- [x] changes.md

## Release Management

* Affected package(s): Surface_mesh_parameterization
* Issue(s) solved (if any): fix #7772
* Link to compiled documentation (obligatory for small feature) [*wrong link name to be changed*](httpssss://wrong_URL_to_be_changed/Manual/Pkg)
* License and copyright ownership: unchanged

albert-github commented on Surface_mesh_parameterization/include/CGAL/Surface_mesh_parameterization/LSCM_parameterizer_3.h:
```diff
@@ -77,10 +77,7 @@ namespace Surface_mesh_parameterization {
-/// \code
-///   OpenNL::SymmetricLinearSolverTraits<typename TriangleMesh::NT>
-/// \endcode
+///
```
It is good to remove the OpenNL.
The lines above the `\code` part of Eigen suggest that when Eigen is no enabled something else is used but what?

albert-github reviewed (commented):
I can see that the file linear_solver.h contains still some  non OpenNL related information.
Why isn't this file moved away from here and isn't the complete OpenNL directory removed?

afabri commented:
> I can see that the file linear_solver.h contains still some non OpenNL related information. Why isn't this file moved away from here and isn't the complete OpenNL directory removed?

That is the first Todo.  Be patient.

sloriot commented:
Successfully tested in CGAL-6.0-Ic-96

lrineau commented:
@afabri This PR is tested, but there is still a TODO entry.
> Currently one file remains, but that fills Eigen matrices and calls an Eigen solver. No need to be in a namespace OpenNL.

Is that something you plan to do?

afabri commented:
> @afabri This PR is tested, but there is still a TODO entry.
> 
> > Currently one file remains, but that fills Eigen matrices and calls an Eigen solver. No need to be in a namespace OpenNL.
> 
> Is that something you plan to do?

yes

albert-github reviewed (commented):
On the risk of being impatient...

I still see the word OpenNL (in different iform of lower / upper case) still in a number of files a.o.:
```
Installation/LICENSE
Maintenance/deb/sid/debian/copyright
Maintenance/deb/squeeze/debian/copyright
Maintenance/deb/wheezy/debian/copyright
Maintenance/infrastructure/cgal.geometryfactory.com/reference-platforms/i686_Linux-2.6_g++-4.1.2_CentOS-5.1/CMakeCache.txt
Maintenance/infrastructure/cgal.geometryfactory.com/reference-platforms/i686_Linux-2.6_g++-4.1.2_CentOS-5.1-O2/CMakeCache.txt
Maintenance/infrastructure/cgal.geometryfactory.com/reference-platforms/i686_Linux-2.6_g++-4.1.2_CentOS-5.1-O3/CMakeCache.txt
Maintenance/infrastructure/cgal.geometryfactory.com/reference-platforms/x86_64_Linux-x.y_IntelCompiler-15.0_CentOS-7.x/CMakeCache.txt
Maintenance/infrastructure/magritte.geometryfactory.com/reference_platforms/x86-64_Darwin-13.0_Apple-clang-5.0_Release/CMakeCache.txt
Maintenance/infrastructure/magritte.geometryfactory.com/reference_platforms/x86-64_Darwin-13.0_Apple-clang-5.0_Release-cpp11/CMakeCache.txt
Maintenance/infrastructure/magritte.geometryfactory.com/reference_platforms/x86-64_Darwin-13.0_Apple-clang-5.1_Release-LEDA-without-GMP/CMakeCache.txt
Maintenance/infrastructure/renoir.geometryfactory.com/reference-platforms/x86-64_Linux-Fedora19_g++-4.5-branch_CXX0X/CMakeCache.txt
Maintenance/infrastructure/renoir.geometryfactory.com/reference-platforms/x86-64_Linux-Fedora19_g++-4.5-branch_Release/CMakeCache.txt
Maintenance/infrastructure/renoir.geometryfactory.com/reference-platforms/x86-64_Linux-Fedora19_g++-4.8/CMakeCache.txt
Maintenance/infrastructure/renoir.geometryfactory.com/reference-platforms/x86-64_Linux-Fedora19_g++-4.8_ansi/CMakeCache.txt
Maintenance/infrastructure/renoir.geometryfactory.com/reference-platforms/x86-64_Linux-Fedora19_g++-4.8_CXX0X/CMakeCache.txt
Maintenance/infrastructure/renoir.geometryfactory.com/reference-platforms/x86-64_Linux-Fedora19_g++-4.8_CXXDEBUG/CMakeCache.txt
Maintenance/infrastructure/renoir.geometryfactory.com/reference-platforms/x86-64_Linux-Fedora19_g++-4.8_m32/CMakeCache.txt
Maintenance/infrastructure/renoir.geometryfactory.com/reference-platforms/x86-64_Linux-Fedora19_g++-4.8_MATCHING-BUG-6/CMakeCache.txt
Maintenance/infrastructure/renoir.geometryfactory.com/reference-platforms/x86-64_Linux-Fedora19_g++-4.8_Release/CMakeCache.txt
Maintenance/infrastructure/renoir.geometryfactory.com/reference-platforms/x86-64_Linux-Fedora19_g++-4.8_Release-LEDA/CMakeCache.txt
Maintenance/infrastructure/renoir.geometryfactory.com/reference-platforms/x86-64_Linux-Fedora19_g++-trunk_CXX0X/CMakeCache.txt
Maintenance/infrastructure/renoir.geometryfactory.com/reference-platforms/x86-64_Linux-Fedora19_IntelCompiler-14.0-with-g++-4.8-STL/CMakeCache.txt
Maintenance/infrastructure/renoir.geometryfactory.com/reference-platforms/x86-64_Linux-Fedora19_IntelCompiler-14.0-with-g++-4.8-STL_strict-ansi/CMakeCache.txt
Maintenance/infrastructure/renoir.geometryfactory.com/reference-platforms/x86-64_Linux-Fedora19_llvm-clang-with-g++-4.8/CMakeCache.txt
Maintenance/infrastructure/renoir.geometryfactory.com/reference-platforms/x86-64_Linux-Fedora_IntelCompiler-17/CMakeCache.txt
Maintenance/infrastructure/renoir.geometryfactory.com/reference-platforms/x86-64_Linux-Fedora_IntelCompiler-17-with-assertions/CMakeCache.txt
Scripts/developer_scripts/check_licenses
Scripts/scripts/cgal_create_CMakeLists
  in comment
Solver_interface/doc/Solver_interface/Solver_interface.txt
Surface_mesh_parameterization/doc/Surface_mesh_parameterization/Surface_mesh_parameterization.txt
  a bit dubious as it is now removed again, maybe an extra remark that it has been removed as of ...
```

afabri commented on Documentation/doc/resources/1.8.13/BaseDoxyfile.in:
```diff
@@ -106,7 +106,6 @@ ALIASES                = "cgal=%CGAL" \
                          "taucs=TAUCS" \
                          "lapack=LAPACK" \
                          "blas=BLAS" \
-                         "opennl=OpenNL" \
                          "cpp=C++" \
```
@sloriot ?

lrineau replied:
done in 36958928dfaaa0721f534b85a2cad3365437ea91

afabri commented:
I left the files in the directory `Maintenance` unchanged.  @lrineau is that ok?

afabri commented:
> Surface_mesh_parameterization/doc/Surface_mesh_parameterization/Surface_mesh_parameterization.txt
>   a bit dubious as it is now removed again, maybe an extra remark that it has been removed as of ...

I added a sentence that it was later removed.

albert-github commented on Surface_mesh_parameterization/doc/Surface_mesh_parameterization/Surface_mesh_parameterization.txt:
```diff
@@ -547,6 +547,7 @@ and replaces a wrapper which had a more complicated API.
 Iterative authalic parameterization was added in CGAL 5.2 by Mael Rouxel-Labbé,
 based on a prototype developed by Hardik Jain.
 
+In CGAL 6.0 the fork of OpenNL that was distributed with CGAL was removed.
+
```
This suggests that it is still possible to use OpenNL, I thought the usage was completely removed.

albert-github replied:
> I thought the usage was completely removed.

New wording:
> In CGAL 6.0 this package no longer uses the fork of OpenNL that was distributed with CGAL
and hence was removed.

Is my assumption about the removal incorrect >
The new wording does not signal my thoughts either.

sloriot commented:
Successfully tested in CGAL-6.0-Ic-116

lrineau commented:
> I left the files in the directory `Maintenance` unchanged. @lrineau is that ok?

I removed them in a later commit (c19cd150cb54d05513fe61720cbe7b194178b5f0).

lrineau commented:
Here are the last occurrences of opennl in CGAL:

```
[lrineau@fernand]~/Git/cgal-master% ack opennl -i  ^build*                                
Installation/CHANGES.md
2477:-   `LSCM_parameterizer_3` now uses by default Eigen instead of OpenNL
4725:    Added Jacobi and SSOR preconditioners to OpenNL solver, which makes

Solver_interface/doc/Solver_interface/Solver_interface.txt
128:on \taucs, \lapack, \blas and OpenNL. Gaël Guennebaud

Surface_mesh_parameterization/doc/Surface_mesh_parameterization/Surface_mesh_parameterization.txt
537:Pierre Alliez and Bruno Lévy. Bruno Lévy added OpenNL support to the package.
550:In CGAL 6.0 this package no longer uses the fork of OpenNL that was distributed with CGAL
```

so:

  - the changelog,
  - and two "history" sections in the manual.

Thas is okay.
